sources/packages/backend/apps/api/src/route-controllers/note/note.institutions.controller.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Query,
+  ParseIntPipe,
+  NotFoundException,
+} from "@nestjs/common";
+import { StudentService } from "../../services";
+import BaseController from "../BaseController";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { NoteType } from "@sims/sims-db";
+import { NoteAPIOutDTO, transformToNoteDTO } from "./models/note.dto";
+import {
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+} from "../../auth/decorators";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { ParseEnumQueryPipe } from "../utils/custom-validation-pipe";
+
+/**
+ * Institution Controller for Notes.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@IsBCPublicInstitution()
+@Controller("note")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-note`)
+export class NoteInstitutionsController extends BaseController {
+  constructor(private readonly studentService: StudentService) {
+    super();
+  }
+
+  /**
+   * Get notes for a student.
+   * @param studentId student id.
+   * @param noteType note type enum which is passed to filter the notes.
+   * @returns student notes.
+   */
+  @ApiNotFoundResponse({ description: "Student not found." })
+  @HasStudentDataAccess("studentId")
+  @Get("student/:studentId")
```

Comment:
I was wondering do we need something like `@Get("student/:studentId/notes")` here. Thoughts?

Reply:
I was trying to check on `Swagger` what we have been doing as a pattern.
Since "note" is already "the resource" I would say that we do not use it but at the same time if we need to provide a note id it would probably be `@Get("student/:studentId/note/:noteId")`